lib/oxidized/model/asa.rb

```diff
@@ -11,11 +11,17 @@ class ASA < Oxidized::Model
 
   cmd :secret do |cfg|
     cfg.gsub! /enable password (\S+) (.*)/, 'enable password <secret hidden> \2'
+    cfg.gsub! /^passwd (\S+) (.*)/, 'passwd <secret hidden> \2'
     cfg.gsub! /username (\S+) password (\S+) (.*)/, 'username \1 password <secret hidden> \3'
     cfg.gsub! /(ikev[12] ((remote|local)-authentication )?pre-shared-key) (\S+)/, '\1 <secret hidden>'
-    cfg.gsub! /^(aaa-server TACACS\+? \(\S+\) host.*\n\skey) \S+$/mi, '\1 <secret hidden>'
+    cfg.gsub! /^(aaa-server TACACS\+? \(\S+\) host[^\n]*\n(\s+[^\n]+\n)*\skey) \S+$/mi, '\1 <secret hidden>'
```

Comment:
This regular expression captures all blocks of the aaa-server, but cuts out secrets only from the latter. If there is no "key" parameter in the block, it can capture any subsequent settings in the configuration.

before:
```
aaa-server TEST1 (CORE) host 192.168.100.1
 timeout 5
 key zzz
aaa-server TEST1 (CORE) host 192.168.100.2
 timeout 5
 key <secret hidden>
```
after:
```
aaa-server TEST1 (CORE) host 192.168.100.1
 timeout 5
 key <secret hidden>
aaa-server TEST1 (CORE) host 192.168.100.2
 timeout 5
 key <secret hidden>
```